src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+          val offset =
+            when (usingGamePiece) {
+              GamePiece.CUBE -> Elevator.TunableElevatorHeights.shelfIntakeCubeOffset.get()
+              GamePiece.CONE -> Elevator.TunableElevatorHeights.shelfIntakeConeOffset.get()
+              else -> 0.0.inches
+            }
+          elevator.currentRequest =
+            Request.ElevatorRequest.TargetingPosition(
+              Elevator.TunableElevatorHeights.doubleSubstationHeight.get() + offset
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition &&
+          manipulator.isAtTargetedPosition &&
+          elevator.isAtTargetedPosition
+        ) {
+          nextState = SuperstructureStates.DOUBLE_SUBSTATION_INTAKE
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.DOUBLE_SUBSTATION_INTAKE -> {
+        // Outputs
+        val gripperVoltage =
+          when (usingGamePiece) {
+            GamePiece.CUBE -> Manipulator.TunableManipulatorStates.cubeInVoltage.get()
+            GamePiece.CONE -> Manipulator.TunableManipulatorStates.coneInVoltage.get()
+            else -> ManipulatorConstants.IDLE_VOLTAGE
+          }
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.doubleSubstationIntakeShelfExtension.get(),
+            gripperVoltage
+          )
+
+        // Transition
+        if (manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.SINGLE_SUBSTATION_INTAKE_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.stowedDownAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.singleSubstationIntakeShelfExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+          val offset =
+            when (usingGamePiece) {
+              GamePiece.CUBE -> Elevator.TunableElevatorHeights.singleSubstationCubeOffset.get()
+              GamePiece.CONE -> Elevator.TunableElevatorHeights.singleSubstationConeOffset.get()
+              else -> 0.0.inches
+            }
+          elevator.currentRequest =
+            Request.ElevatorRequest.TargetingPosition(
+              Elevator.TunableElevatorHeights.singleSubstationHeight.get() + offset
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition &&
```

Comment:
same question as above -- what happens if `isAtTargetedPosition` changes value between the previous `if` and here?